test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_enqueue.cpp

```diff
@@ -0,0 +1,563 @@
+                                          nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clEnqueueCommandBufferKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        queue1 = clCreateCommandQueue(context, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+        queue2 = clCreateCommandQueue(context, device, 0, &error);
+        test_error(error, "clCreateCommandQueue failed");
+
+        return CL_SUCCESS;
+    }
+
+    clCommandQueueWrapper queue1;
+    clCommandQueueWrapper queue2;
+};
+
+// CL_INVALID_COMMAND_QUEUE if any element of queues is not a valid
+// command-queue.
+struct EnqueueCommandBufferNotValidQueueInQueues : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_command_queue queues[1] = { nullptr };
+        cl_int error = clEnqueueCommandBufferKHR(1, queues, command_buffer, 0,
+                                                 nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clEnqueueCommandBufferKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INCOMPATIBLE_COMMAND_QUEUE_KHR if any element of queues is not compatible
+// with the command-queue set on command_buffer creation at the same list index.
+struct EnqueueCommandBufferQueueNotCompatible : public BasicCommandBufferTest
+{
+    EnqueueCommandBufferQueueNotCompatible(cl_device_id device,
+                                           cl_context context,
+                                           cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue),
+          queue_not_compatible(nullptr)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int error = clEnqueueCommandBufferKHR(
+            1, &queue_not_compatible, command_buffer, 0, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INCOMPATIBLE_COMMAND_QUEUE_KHR,
+                               "clEnqueueCommandBufferKHR should return "
+                               "CL_INCOMPATIBLE_COMMAND_QUEUE_KHR",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        queue_not_compatible = clCreateCommandQueue(
+            context, device, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, &error);
```

Comment:
`CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE` is an optional property, so `CL_QUEUE_PROFILING_ENABLE` would be a better property to use here so that we don't need to query the device first for support.